### PR TITLE
[DMA 5/8]: Clean up macros

### PR DESCRIPTION
**Pull request by bugadani**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

#### Description

This PR mostly cleans up macros and interrupt handlers (reducing the number of locations we need to edit), but also refactors the PDMA macro a bit in preparation for CryptoDMA support.

#### Testing
Describe how you tested your changes.

### CODE REVIEW

**Comment by Dominaezzz** [comment]
fwiw, the last commit LGTM. (Idk if it's easier to get this in earlier to avoid rebasing it)

**Comment by bugadani** [comment]
I'm rebasing this PR over and over again so that I can have an overview of what I'm missing. The idea of the macro dedup came a bit too late, I'd rather not pull it up and rebase over it, especially since it doesn't matter that much.

File: esp-hal/src/dma/pdma.rs [hunk 1]
```diff
@@ -857,23 +749,24 @@ impl InterruptAccess<DmaRxInterrupt> for AnyI2sDmaRxChannel {
...
 
-macro_rules! ImplI2sChannel {
-    ($num: literal) => {
+macro_rules! ImplPdmaChannel {
+    ($peri:ident, $num:literal, $int:ident) => {
         paste::paste! {
-            #[doc = concat!("DMA channel suitable for I2S", $num)]
-            pub struct [<I2s $num DmaChannel>] {}
+            #[doc = concat!("DMA channel suitable for ", stringify!([< $peri:upper >]), $num)]
+            #[non_exhaustive]
```
Comment (bugadani) [inline esp-hal/src/dma/pdma.rs:756]: I guess this bug has lived long enough, no more safely constructed I2S DMA channels 🙃

Comment (bugadani) [reply esp-hal/src/dma/pdma.rs:756]: Not anymore, these are the channel "peripherals".

File: esp-hal/src/macros.rs [hunk 2]
```diff
@@ -134,3 +134,13 @@ macro_rules! if_set {
         $set
     };
 }
+
+/// Macro to ignore tokens.
+///
+/// This is useful when we need existence of a metavariable (to expand a
+/// repetition), but we don't need to use it.
```
Comment (Dominaezzz) [inline esp-hal/src/macros.rs:141]: Worth adding the FIXME comment here as well?

**Review by Dominaezzz — APPROVED** [review]

File: esp-hal/src/macros.rs (hunk 2 above)
Comment (bugadani) [reply esp-hal/src/macros.rs:141]: In fact one here would be sufficient, but it's fine and I'd not poke at this PR again just for this.

**Pull request merged**